### DISCUSSION TITLE
I made a github ci pipeline that uses this project to install k8s on hetzner cloud

### DISCUSSION THREAD

**Discussion by vilaca** [kube-hetzner/terraform-hcloud-kube-hetzner#1197] [Show and tell]
Fell free to test, contribute or ask any questions. Suggestions on how to improve are greatly appreciated.

The repository is https://github.com/vilaca/iac-hcloud-kubernetes

**Comment by codeagencybe** [comment]
Can you elaborate a bit more what it does exactly? 
I understand it uses Github CI, probably as a GH action? But what is the advantage? 
Does it store the state of the cluster inside Github and each change triggers the GH action, build and CICD pipeline again?

**Comment by vilaca** [comment]
This is the 1st step of a learning project i intend on continuing in the next months (depending on my availability to work on this).

Right now it does:
- the same as running terraform-hcloud-kube-hetzner from the command line (build os images, install k8s)
- installs a whoami http service with ingress (for a quick sanity check)
- sends an email with the kubeconfig.yaml, the generated pvt/public keys and a link for the http service mentioned above

<img width="895" alt="image" src="https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/assets/38896/e01f8161-e87f-4aa3-836e-c801968047b8">

Right now it is a toy/poc but if time allows it I hope to make it production ready. Thank you for your interest.
